File: extract-resume.js
const fs = require('fs');
const path = require('path');
const pdfjsLib = require('pdfjs-dist');

const pdfPath = path.join(__dirname, 'src/assets/Resume.pdf');

async function extractPdf() {
  try {
    const dataBuffer = fs.readFileSync(pdfPath);
    const uint8Array = new Uint8Array(dataBuffer);
    const pdf = await pdfjsLib.getDocument({ data: uint8Array }).promise;
    let text = '';

    for (let i = 1; i <= pdf.numPages; i++) {
      const page = await pdf.getPage(i);
      const content = await page.getTextContent();
      const pageText = content.items.map((item) => item.str).join(' ');
      text += pageText + '\n';
    }

    console.log('=== RESUME TEXT CONTENT ===\n');
    console.log(text);
  } catch (err) {
    console.error('Error:', err.message);
    process.exit(1);
  }
}

extractPdf();

File: extract_resume.py
#!/usr/bin/env python3
import sys
import subprocess

# First, try to import PyPDF2, if not available, install it
try:
    import PyPDF2
except ImportError:
    print("Installing PyPDF2...")
    subprocess.check_call([sys.executable, "-m", "pip", "install", "PyPDF2", "-q"])
    import PyPDF2

pdf_path = r"c:\Users\mdspe\projects\Portfolio Website\ai-tool-nextjs-main\src\assets\Resume.pdf"

try:
    with open(pdf_path, 'rb') as file:
        reader = PyPDF2.PdfReader(file)
        print("=== RESUME TEXT CONTENT ===\n")
        
        for page_num in range(len(reader.pages)):
            page = reader.pages[page_num]
            text = page.extract_text()
            print(text)
            print("\n--- Page {} end ---\n".format(page_num + 1))
            
except Exception as e:
    print(f"Error: {e}", file=sys.stderr)
    sys.exit(1)
